### PR TITLE
Implement and test more aggressive compacting for WAF regexes

### PR DESCRIPTION
**Pull request by dannyroberts**
https://dimagi-dev.atlassian.net/browse/SAAS-12315?focusedCommentId=229359

##### ENVIRONMENTS AFFECTED
staging, india, production

### CODE REVIEW

**Review by gherceg — APPROVED** [review]
Minor suggestion but besides that looks good.

File: src/commcare_cloud/commands/terraform/terraform.py [hunk 1]
```diff
@@ -187,15 +188,59 @@ def generate_terraform_entrypoint(environment, key_name, run_dir, apply_immediat
             f.write(render_template(template_file, context, template_root))
 
 
-def compact_waf_regexes(patterns):
+def compact_waf_regexes(patterns, compactible_affixes=None):
+    compactible_affixes = COMPACTIBLE_AFFIXES if compactible_affixes is None else compactible_affixes
```
[the remark this answers is not in the document]
Comment (dannyroberts) [reply src/commcare_cloud/commands/terraform/terraform.py:196]: It's not a big deal, but I did this intentionally. Basically if I use `or` then that makes it impossible to pass in `[]` as a value here, which is a conceivable value ("don't use any compactible affixes") distinct from the way I'm using `None` here ("use the default list"). I haven't actually used any value for this other than the default in code or tests, but if someone did pass in `[]` (in, say, a test) I'd want that interpreted differently from `None`.

[the remark this answers is not in the document]
Comment (gherceg) [reply src/commcare_cloud/commands/terraform/terraform.py:196]: Ah ok that makes sense. Thanks for the explanation!

File: src/commcare_cloud/commands/terraform/tests/test_compact_regexes.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+def test_compact_waf_regexes_simply__match_examples():
+    _test_compact_function_against_examples(compact_waf_regexes_simply, COMMCAREHQ_XML_POST_URLS_REGEX)
+
+
+def test_compact_waf_regexes__single_pattern():
+    assert compact_waf_regexes([r'^/a/([\w\.:-]+)/api/v([\d\.]+)/form/$']) \
+           == [r'^/a/([\w\.:-]+)/(api/v([\d\.]+)/form/)$']
+
+
+def test_compact_waf_regexes_simply__two_patterns():
```
Comment (AmitPhulera) [inline src/commcare_cloud/commands/terraform/tests/test_compact_regexes.py:30]: I guess this test is the same as https://github.com/dimagi/commcare-cloud/pull/5525/files#diff-ddbd1b76aaed0c6cb3a6edb4f653f7d341bd3dbca2aa06091ad30fba48cd3dc9R12?

I am wondering why stickler did not complain about it. 🤔

**Review by AmitPhulera — APPROVED** [review]

**Comment by AmitPhulera** [comment]
FYI @dannyroberts Test failure looks related.

**Comment by dannyroberts** [comment]
Thanks, it was related to new things in the merge. Should be fixed now

**Review by AmitPhulera — APPROVED** [review]

**Pull request merged**